classes/class-newsmatchdonation-shortcode.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * The class for the News Match Donation Shortcode, and associated functions
+ *
+ * @package NewsMatchDonation\Shortcode
+ */
+
+/**
+ * Register a shortcode
+ */
+class NewsMatchDonation_Shortcode {
+	/**
+	 * The prefix used for this plugin's options saved in the options table
+	 *
+	 * @var string $options_prefix The prefix for this plugin's options saved in the options table
+	 */
+	protected $option_prefix = '';
+
+	/**
+	 * The constructor
+	 */
+	public function __construct() {
+		$this->option_prefix = NewsMatchDonation_Settings::$options_prefix;
+
+		add_action( 'wp_enqueue_scripts', array( $this, 'register_assets' ) );
+		add_shortcode( 'newsmatch_donation_form', array( $this, 'donation_form_shortcode' ) );
+	}
+
+	/**
+	 * Define the donation form shortcode.
+	 * <p>
+	 *     This shortcode is intended for use somewhere on a page.
+	 * </p>
+	 * <p>
+	 *     Example usage:
+	 *     Add donation form with no Salesforce campaign id and no default donation amount specified:
+	 *     [newsmatch_donation_form]
+	 *
+	 *      Add donation form with no Salesforce campaign id and $50.00 as the default donation amount:
+	 *      [newsmatch_donation_form amount="50"]
+	 *
+	 *      Add a donation form with a Salesforce campaign id of "foo" and $25.00 as the default donation amount:
+	 *      [newsmatch_donation_form sf_campaign_id="foo" amount="25"]
+	 *
+	 *      Add a donation form with a Salesforce campaign id of "foo" and do not specify a default donation amount:
+	 *      [newsmatch_donation_form sf_campaign_id="foo"]
+	 * </p>
+	 *
+	 * @param  array $atts The attribute values passed in through the shortcode.
+	 * @return string The HTML markup for the donation form.
+	 */
+	public function donation_form_shortcode( $atts ) {
+		if ( isset( $atts['type'] ) && 'select' === $atts['type'] ) {
+			return $this->render_view( '/views/rr-donation-form-select.view.php', $atts );
+		} else {
+			return $this->render_view( '/views/rr-donation-form-buttons.view.php', $atts );
+		}
+	}
+
+	/**
+	 * Register the donation plugin's shortcode's CSS and Javascript files.
+	 */
+	public function register_assets() {
+		wp_register_style(
+			'newsmatch-donation',
+			plugins_url( 'assets/css/donation.css', NMD_PLUGIN_FILE )
+		);
+
+		wp_enqueue_style( 'newsmatch-donation' );
+
+		wp_register_script(
+			'newsmatch-donation',
+			plugins_url( 'assets/js/donation.js', NMD_PLUGIN_FILE ),
+			array( 'jquery' ),
+			null,
+			true
+		);
+
+		wp_enqueue_script( 'newsmatch-donation' );
```

Comment:
If you're registering and enqueueing a style or script all at once you can just use the enqueue function rather than registering first then enqueuing. The register / enqueue separation is useful if you're looking to register a style or script now to be registered at a future time (usually under conditional circumstances).